### PR TITLE
turning in my assessment

### PR DESCRIPTION
**Pull request by roeckerc**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x ] Coding Challenges
- [ x] Coding Concepts
- [x ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -12,58 +12,153 @@
...
 const object2 =  { number: 0 }
 // Expected output: "0 is divisible by three"
 const object3 = { number: -7 }
 // Expected output: "-7 is not divisible by three"
 
+// create a checker  with a funciton called "demarcateThrice", it checks to see if the value of a key of an object is devisable by 3.
+// later when i write the function i can check if its devisible by 3 by using % 3 and looking to see if it strictly equals zero. 
 
+
+describe("demarcateThrice", () => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:55]: test structure look nice as well as indentation

File: code-challenges.test.js [hunk 2]
```diff
@@ -12,58 +12,153 @@
...
+    it("checks to see if the value of a key of an object is devisable by 3", () => {
+      expect(demarcateThrice(object1.number)).toEqual(`15 is divisible by three`);
```
Comment (SunkissedQueen) [inline code-challenges.test.js:57]: you definitiely need dot notation to assess the value. However this should be placed in your function. The idea of the expect statement is shows a function call with an argument and the expected output. Logic is handled in the function

Comment (SunkissedQueen) [reply code-challenges.test.js:57]: so best practice is to instead have the following expect statement
`expect(demarcateThrice(object1)).toEqual("15 is divisible by three")`

File: code-challenges.test.js [hunk 3]
```diff
@@ -12,58 +12,153 @@
...
+    });
+  });
+//returned 1 fail.
 // b) Create the function that makes the test pass.
+// create a funciton called "demarcateThrice", it checks to see if the value of a key of an object is devisable by 3.
+// i can check if its devisible by 3 by using % 3 and looking to see if it strictly equals zero. 
+// i can then have it return the statement, starting with the number that it used, then say if it is of is not divisible by three.
+
+
+const demarcateThrice = (number) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:69]: this structure is great. You will just have to use the dot notation from earlier. Ex:
`const demarcateThrice = (object) => {
    if (object.number %3 === 0) {
        return `${object.number} is divisible by three`
    }`
It is hard to portray the exact syntax in this comment section. Reach out if you need clarity on these explanations

File: code-challenges.test.js [hunk 4]
```diff
@@ -12,58 +12,153 @@
...
+    if (number %3 === 0) {
+        return `${number} is divisible by three`
+    } else {
+        return `${number} is not divisible by three`
+    }
+}
```
Comment (SunkissedQueen) [inline code-challenges.test.js:76]: Did your function pass?

File: code-challenges.test.js [hunk 5]
```diff
@@ -12,58 +12,153 @@
...
+    // declare a function that takes in an array
+    // create a local variable that maps through the array
+    // with each value access the first index and capitalize the letter then concatenate the remainder of the word
+    // return the local variable joined into a string
+    const firstLetterFertalizer = (array) => {
+        
+      let eachItem = array.map((value) => {
+        
+          
+          return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
```
Comment (SunkissedQueen) [inline code-challenges.test.js:105]: I like that you considered examples outside of the edge cases of the test variables.

File: code-challenges.test.js [hunk 6]
```diff
@@ -12,58 +12,153 @@
...
 
 // a) Create a test with expect statements for each of the variables provided.
-
+describe("firstLetterFertalizer", () => {
+    it("takes the first letter and capitalizes it", () => {
+      expect(firstLetterFertalizer(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]);
+      expect(firstLetterFertalizer(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]);
+    });
+  });
+//returned 1 fail
```
Comment (SunkissedQueen) [inline code-challenges.test.js:88]: good, like your function name

File: code-challenges.test.js [hunk 7]
```diff
@@ -12,58 +12,153 @@
...
+        
+      let eachItem = array.map((value) => {
+        
+          
+          return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
+      });
+   
+       return eachItem;
+      }
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:110]: did your function pass?

File: code-challenges.test.js [hunk 8]
```diff
@@ -12,58 +12,153 @@
...
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
-
+//pseudocode: make a function
+// look for vowels and log the index of them if non log -1.
+// get rid of all the index less than 0
+// organize smallest to largest putting the smallest number in the 0 index of new array.
+//get the 0 index which is the first location of the vowel.
+// i spent like 4 hours on this and it was way easier than i made it cause i was stupid. see practice.js to see the pain. i deleted so much and was still left with all that dumb stuff hahahaha. wow this one rocked me. I really wish i did the concepts and interview questions first, it would have made this wayyyyy easier.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:138]: look like a case of thinking-it-has-to-be-complicated mentality. I recommend searching for javascript built-in methods for strings. You will see that you can use several less lines to accomplish the mission.

File: coding-concepts.js [hunk 9]
```diff
@@ -2,72 +2,95 @@
...
+// const cohort = "Echo 2022"
+// // console.log(cohort.split(""))
 
-// --------------------2) What will this log?
+// // a) Your answer: This will log the array: 
+//      ['E', 'c', 'h', 'o', ' ', '2', '0', '2', '2']
+// // b) Verify and explain:
+// I was correct. .Split is an Accessor. Accessor methods dont actually change the array. They will output something, but if you were to just console.log(cohort), it would still be = "Echo 2022". If it wasnt going straight into a concole log, you would have to save it as a varible ex: var imSoTired = cohort.split("").
+// I got super lucky on guessing how it would split it up. The place you have the "'s when using .split will change the answer ex:
+// if it had been console.log(cohort.split(" ")) the reuslt would have benn ["Echo", "2022"]
```
Comment (SunkissedQueen) [inline coding-concepts.js:21]: glad you look beyond the question. Have you tried placing emoji's instead of a space?

File: coding-concepts.js [hunk 10]
```diff
@@ -2,72 +2,95 @@
...
+// // a) Your answer: This will log "undefined". Every function must have a return. 
+// // b) Verify and explain: The parts of a function are
+// 1 variable declaration
+// Variable name
+// Single equal sign
+// Parentheses
+// Arrow
+// Curly Braces
+// Return 
+// without parts of this, it would not be a function. There is a thing called syntactical sugar that you could use, but the concept is still the same. A function is a recipe, but you have to have something that comes out at the end or else the recipe is pointless. The return is necessary.
```
Comment (SunkissedQueen) [inline coding-concepts.js:41]: nice notes

File: coding-concepts.js [hunk 11]
```diff
@@ -2,72 +2,95 @@
...
-// b) Verify and explain:
+// // --------------------3) What will this log?
 
+// const multipliedByTwo = [4, 5, 6, 7, 8].map(number => number * 2)
+// // console.log(multipliedByTwo)
 
-// --------------------4) What will this log?
+// // a) Your answer: [8, 10, 12, 14, 16].  
+// // b) Verify and explain: line 45 is declaring a variable named multipliedByTwo. multipliedByTwo = and array that is being mapped over with a function. .map is a higher order function that goes over each part of an array and does or check something to each item at an index.
+// In this case it is mapping over the array and multiplying each number by 2 and returning a new array that is the same length.
```
Comment (SunkissedQueen) [inline coding-concepts.js:50]: exactly

File: coding-concepts.js [hunk 12]
```diff
@@ -2,72 +2,95 @@
...
-// console.log(onlyOdds)
+// // --------------------4) What will this log?
 
-// a) Your answer:
-// b) Verify and explain:
+// const onlyOdds = [11, 12, 13, 14, 15].filter(number => number % 2 !== 0)
+// // console.log(onlyOdds)
 
+// // a) Your answer:[11, 13, 15]
+// // b) Verify and explain: line 54 is declaring a variable name onlyOdds. onlyOdds = an array that is being filtered over with a function. .filter is a higher order function, alot like .map, but it goes over each part of an array and does or checks something to each item at an index, ,but it does not have to return an array of the same length. it will only return the things that were true with the condition set.
```
Comment (SunkissedQueen) [inline coding-concepts.js:58]: good explanation

File: coding-concepts.js [hunk 13]
```diff
@@ -2,72 +2,95 @@
...
+// const myCodingSkills = {
+//   languages: ["JavaScript", "Ruby"],
+//   frameworks: ["React", "Ruby on Rails"],
+//   databases: "PostgreSQL",
+//   versionControl: "GitHub"
+// }
+// // console.log(myCodingSkills.languages[0])
 
+// // a) Your answer: JavaScript
+// // b) Verify and explain: i was right. What we have here is an object. the object is myCoding Skills. Objects have keys and value pairs. keys and value are the 2 parts to an object. In this example, the keys are languages, frameworks, databases, and versionControl. they are all a datatype of symbol. the value of each of the keys is to the right of them.
```
Comment (SunkissedQueen) [inline coding-concepts.js:73]: good walk through

File: coding-concepts.js [hunk 14]
```diff
@@ -2,72 +2,95 @@
...
+//   constructor(name){
+//     this.student = name,
+//     this.cohort = "Echo",
+//     this.year = 2022
+//   }
+// }
+// const learnStudent = new Learn("George")
+// // console.log(learnStudent)
+
+// // a) Your answer: Learn { student: 'George', cohort: 'Echo', year: 2022 }
```
Comment (SunkissedQueen) [inline coding-concepts.js:89]: very impressed

File: interview-questions.md [hunk 15]
```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: a parameter is like what we are acting on, and the arguement is what we are saying for that thing to do or check
 
 Researched answer:
 
+A Parameter is a "placeholder value". The information that goes there is not available at this time, but when it is needed, it will be supplied and you can put it right there in the parameter spot. it can also be referenced multiple times in the function. When that parameter is filled. it is filled with the argument. An argument is whatever filled up that "placeholder value"
```
Comment (SunkissedQueen) [inline interview-questions.md:15]: I think I understand what you are saying, just need some clarification. Parameters are place holders while arguments are values. Parameters allow you to dynamically accept any values as arguments.

File: interview-questions.md [hunk 16]
```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+An Argument is contained in a set of (), which are a parameter. Its job is to be the value that is being referenced.  A function or method may not need and argument in which case, it could just be a set of empty () which is a parameter. Sometimes some functions will have multiple parameters, which will need multiple arguments to fills those spots. 
+
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer:The Javascript built in method .map() is a higher order function that filters through an array, and perfroms an action on each value at an index, and  returns an array of the same length. The predefined parameters that .map takes are as follows
+"car port" is what is being mapped
+("") returns each charater of the string in " includeing the space ex ["c", "a", "r", " ", "p", "o", "r", "t"]
```
Comment (SunkissedQueen) [inline interview-questions.md:22]: your example looks like you were outputting the split("") method. Please provide an example using an array to showcase this array built-in method. Never mind, I see that you caught this error. I am just leaving this words because it took time to type them.

File: interview-questions.md [hunk 17]
```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 Researched answer:
+Document Object Model: it is a layout of how a page is loaded. It is like a tree diagram and it makes each part of the page an object so that JS can change everything if it wants to. 
 
 5. What is React? Why would you use it?
 
-Your answer:
+Your answer: React, created by Facebook, is a library program used to build web pages for different platforms and user interfaces. It is super cool. It makes it so that certain parts of the page can change and the whole page does not have to be reloaded.
 
 Researched answer:
+ React, created by Facebook, is a library program used to build web pages for different platforms and user interfaces. It is super cool. It can do a ton of stuff. It is designed to be a ton of little things that join together to make a big thing and make it function well. For instance you can set it up so it makes certain parts of the page  change, like clicking a like button and getting a thumbs up,  and the whole page does not have to be reloaded. React also makes it so the code writer does not have to rewrite code, but instead can just reference old code and have it act upon new information, kind of like assigning a var name to a function.
```
Comment (SunkissedQueen) [inline interview-questions.md:63]: like the reassigning a variable analogy

File: interview-questions.md [hunk 18]
```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
+const evenSteven = (array) => {
+  return array.filter((value) => value % 2 === 0)
+}
+
+console.log(evenSteven(howDoesThisWork))
+
+it returns [8, 10]
+Researched answer: I do think my answer was pretty spot on
```
Comment (SunkissedQueen) [inline interview-questions.md:49]: though your answer is great, there is always room to learn and share something different.

File: interview-questions.md [hunk 19]
```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 Researched answer:
+ React, created by Facebook, is a library program used to build web pages for different platforms and user interfaces. It is super cool. It can do a ton of stuff. It is designed to be a ton of little things that join together to make a big thing and make it function well. For instance you can set it up so it makes certain parts of the page  change, like clicking a like button and getting a thumbs up,  and the whole page does not have to be reloaded. React also makes it so the code writer does not have to rewrite code, but instead can just reference old code and have it act upon new information, kind of like assigning a var name to a function.
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: no clue
 
-Researched answer:
+Researched answer: the idea is that even if the var is not yet declared that it can be refernced prior in the code, unless you are using let or const. But i 100% do not understand this at all. i do not understand the "initialize" thing.
```
Comment (SunkissedQueen) [inline interview-questions.md:69]: hoisting is easier to learn by practicing it

File: interview-questions.md [hunk 20]
```diff
@@ -8,50 +8,80 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 ## Looking Ahead: Terms for Next Week
```
Comment (SunkissedQueen) [inline interview-questions.md:72]: hopefully looking up the terms will help you connect with the material this week

**Review by SunkissedQueen — APPROVED** [review]
Cliff, another assessment session of laughter and good explanations. Here are some tips to show progress for next week:

In the coding challenges, your test structure is nice. Please make sure you go back to look at redundant code in your function. The purpose for a function is so you don't have to repeat code. 
To level up in the interview questions. Remember to use the research not only to ensure your answer is correct but also to learn something different about the topic.
